paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -258,16 +261,50 @@ def paasta_mark_for_deployment(args):
     return ret
 
 
-def instances_deployed(cluster, service, instances, git_sha):
-    result = []
+def instances_deployed(cluster, service, instances_in, instances_out, git_sha, green_light):
+    """Create a thread pool to run _run_instance_worker()
+
+    Args:
+        green_light: an instance of threading.Event(). It is used to terminate
+            all running threads after KeyboardInterrupt.
+        instances_in (Queue): a thread-safe queue that contains all cluster
+            instances that need to be checked.
+        instances_out (Queue): a empty thread-safe queue. I will contain
+            instances that are not deployed yet.
+    """
+    num_threads = min(5, instances_in.qsize())
```

Comment:
This limits the number of simultaneous requests to a cluster.
Otherwise if a service had 100 instances it would query all 100 at once, because they all would be in the instances_in queue.